src/ast.js

```diff
@@ -254,22 +253,26 @@ AST.prototype.resolvePrecedence = function(result, parser) {
       }
     }
   } else if (
-    result.kind === "cast" &&
-    result.what &&
-    !result.what.parenthesizedExpression
+    (result.kind === "cast" &&
+      result.what &&
+      !result.what.parenthesizedExpression) ||
+    (result.kind === "silent" &&
+      result.expr &&
+      !result.expr.parenthesizedExpression)
   ) {
     // https://github.com/glayzzle/php-parser/issues/172
-    if (result.what.kind === "bin") {
-      buffer = result.what;
-      result.what = result.what.left;
-      this.swapLocations(result, result, result.what, parser);
+    const subject = result.kind === "cast" ? "what" : "expr";
+    if (result[subject].kind === "bin") {
+      buffer = result[subject];
+      result[subject] = result[subject].left;
+      this.swapLocations(result, result, result[subject], parser);
       buffer.left = this.resolvePrecedence(result, parser);
       this.swapLocations(buffer, buffer.left, buffer.right, parser);
       result = buffer;
-    } else if (result.what.kind === "retif") {
-      buffer = result.what;
-      result.what = result.what.test;
-      this.swapLocations(result, result, result.what, parser);
+    } else if (result[subject].kind === "retif") {
```

Comment:
the last `...if (result.what.kind === "retif"` becomes dead code and should be removed : 
https://coveralls.io/builds/27873326/source?filename=src/ast.js#L291

Reply:
I removed it but I must admit that I don't really see how my change caused this. Can you explain it?

Reply:
before there was duplicated code between `what` and `expr` properties, as your code is more generic here : https://coveralls.io/builds/27873326/source?filename=src/ast.js#L272 the last elseif check does never trigger

Reply:
Alright! :+1: